### PR TITLE
Obsolete variables removed/Autotuner variable =to?= updated

### PR DESCRIPTION
**Pull request by vijayank88**
Signed-off-by: vijayank88 <paruthi143@gmail.com>

Part of https://github.com/The-OpenROAD-Project/OpenROAD/pull/1479 this PR, @maliberty @vvbandeira  reviewed and wants clean-up for OpenROAD-flow-scripts platform config.mk.
Some of variables values changed to ?= from = (Re-assignable variables.)

modified:   flow/platforms/asap7/config.mk
Changes: Obsolete variables Removed
```
MAX_FANOUT
MAX_WIRE_LENGTH
CTS_MAX_SLEW
CTS_MAX_CAP
CTS_SQR_CAP
 CTS_SQR_RES   
 CTS_SLEW_INTER
 CTS_CAP_INTER 
Variables re-aligned with functional usage order
```
modified:   flow/platforms/nangate45/config.mk
Obsolete variable removed
```
MAX_FANOUT
MAX_WIRE_LENGTH
CTS_MAX_SLEW
CTS_MAX_CAP
```
modified:   flow/platforms/sky130hd/config.mk
Obsolete variable removed
```
MAX_FANOUT
MAX_WIRE_LENGTH
CTS_MAX_SLEW
CTS_MAX_CAP
```
modified:   flow/platforms/sky130hs/config.mk
Obsolete variable removed
```
MAX_FANOUT
MAX_WIRE_LENGTH
```

### CODE REVIEW

**Comment by vijayank88** [comment]
@maliberty  
Without `ABC_CLOCK_PERIOD_IN_PS` variable, chameleon running fine for macro placement. If include ABC_CLOCK_PERIOD its fails for macro placement.
Here is my findings:
1. Without ABC_CLOCK_PERIOD_IN_PS:
```
[INFO][FLOW-xxxx] Using manual macro placement file ./platforms/sky130hd/chameleon/chameleon.macro_placment.cfg
_c869220600000000_p_odb__dbBlock
1000
file7
LINE : _3252_ R0 200 150
LINE : _3253_ R0 1600 150
LINE : _3254_ R180 1289.845 3339.36
LINE : ahb_sys_0_uut.S0.CACHE R0 2100 2650
LINE : ibex_core R0  2150 1700
LINE : ahb_sys_0_uut.apb_sys_inst_0 R0 1470 2643
LINE : 
```
2. With ABC_CLOCK_PERIOD_IN_PS:
```
[INFO ODB-0134] Finished DEF file: ./results/sky130hd/chameleon/base/2_3_floorplan_tdms.def
[INFO][FLOW-xxxx] Using manual macro placement file ./platforms/sky130hd/chameleon/chameleon.macro_placment.cfg
_389be40600000000_p_odb__dbBlock
1000
file7
LINE : _3252_ R0 200 150
Error: macro_place.tcl, 145 Cannot find instance _3252_
```
There was issue with `ord::get_db_block`. 
Is there any fix to get right db_block or proceed with commenting `ABC_CLOCK_PERIOD_PS` in sky130hd config.mk to pass this CI?

**Comment by maliberty** [comment]
The CI fails. I believe @vvbandeira suggested to push the cleanup to a separate PR as it looks to need extra work.  I'm ok with that.

**Comment by rovinski** [comment]
`IO_PIN_MARGIN` is deprecated and can be removed.

**Comment by vijayank88** [comment]
@rovinski 
Thanks for pointing `IO_PIN_MARGIN`. Variable removed across sky130/nangate45/asap7 platform config.mk

@maliberty 
For sky130 alone `ABC_CLOCK_PERIOD_IN_PS` commented to pass CI and created new issue with subject
```
Can't find instance if use ABC_CLOCK_PERIOD_IN_PS for sky130hd chameleon
```
and assigned you.

File: flow/platforms/sky130hd/config.mk [hunk 1]
```diff
@@ -80,7 +78,8 @@ export ADDER_MAP_FILE ?= $(PLATFORM_DIR)/cells_adders_hd.v
 # Define ABC driver and load
 export ABC_DRIVER_CELL = sky130_fd_sc_hd__buf_1
 export ABC_LOAD_IN_FF = 5
-#export ABC_CLOCK_PERIOD_IN_PS = 10
+# Set yosys-abc clock period to first "-period" found in sdc file
+#export ABC_CLOCK_PERIOD_IN_PS ?= $(shell grep -E -o -m 1 "\-period\s+\S+" $(SDC_FILE) | awk '{print $$2}')
```
Comment (maliberty) [inline flow/platforms/sky130hd/config.mk]: Just remove it

**Review by maliberty — APPROVED** [review]

**Pull request merged**